<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>send_messege</class>
 <widget class="QDialog" name="send_messege">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1021</width>
    <height>661</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1021</width>
    <height>718</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <widget class="QWidget" name="layoutWidget1">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>110</y>
     <width>916</width>
     <height>404</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="componate_img_center">
    <item>
     <layout class="QVBoxLayout" name="componate_img_original_with_all">
      <item>
       <widget class="QLabel" name="lbl_img_original">
        <property name="text">
         <string>Исходное изображение</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="img_original">
        <property name="minimumSize">
         <size>
          <width>350</width>
          <height>350</height>
         </size>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="componate_button_load">
        <item>
         <widget class="QPushButton" name="button_choose_img">
          <property name="text">
           <string>Выбрать</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="button_make_img">
          <property name="text">
           <string>Подобрать</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="componate_buttons_crypto_2">
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Операции с шифрами</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="componate_buttons_crypto">
        <item>
         <widget class="QPushButton" name="button_crypto_p">
          <property name="minimumSize">
           <size>
            <width>100</width>
            <height>50</height>
           </size>
          </property>
          <property name="text">
           <string>Зашифровать Р</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="button_crypto_s">
          <property name="minimumSize">
           <size>
            <width>100</width>
            <height>50</height>
           </size>
          </property>
          <property name="text">
           <string>Зашифровать S</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="button_crypto_cansel">
          <property name="minimumSize">
           <size>
            <width>150</width>
            <height>50</height>
           </size>
          </property>
          <property name="text">
           <string>Отменить шифр</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPushButton" name="button_crypt_rsa">
        <property name="minimumSize">
         <size>
          <width>150</width>
          <height>50</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>150</width>
          <height>50</height>
         </size>
        </property>
        <property name="text">
         <string>Зашифровать RSA</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="button_cancel_rsa">
        <property name="minimumSize">
         <size>
          <width>150</width>
          <height>50</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>150</width>
          <height>50</height>
         </size>
        </property>
        <property name="text">
         <string>Отменить RSA</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="componate_img_changed">
      <item>
       <widget class="QLabel" name="lbl_img_changed">
        <property name="text">
         <string>Изображение после преобразований</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="img_changed">
        <property name="minimumSize">
         <size>
          <width>350</width>
          <height>350</height>
         </size>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QPushButton" name="button_send_messege">
   <property name="geometry">
    <rect>
     <x>740</x>
     <y>590</y>
     <width>131</width>
     <height>61</height>
    </rect>
   </property>
   <property name="text">
    <string>Отправить сообщение</string>
   </property>
  </widget>
  <widget class="QWidget" name="layoutWidget_2">
   <property name="geometry">
    <rect>
     <x>530</x>
     <y>530</y>
     <width>261</width>
     <height>56</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="componate_button_algoritme">
    <item>
     <widget class="QPushButton" name="button_algoritm_crypto">
      <property name="text">
       <string>Зашифровать исходное сообщение по алгоритму</string>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="componate_button_algoritme_delte">
      <item>
       <widget class="QPushButton" name="button_algoritm_crypto_delete">
        <property name="text">
         <string>Сбросить алгоритм</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_5">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="layoutWidget_4">
   <property name="geometry">
    <rect>
     <x>720</x>
     <y>0</y>
     <width>298</width>
     <height>104</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="componate_show_now_use_all">
    <item>
     <layout class="QHBoxLayout" name="componate_lbl_show_now_use">
      <item>
       <widget class="QLabel" name="lbl_show_now_use">
        <property name="text">
         <string>Сейчас используются ключи:</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QGridLayout" name="componate_show_now_use_info">
      <item row="0" column="1">
       <widget class="QLabel" name="lbl_show_value">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>20</height>
         </size>
        </property>
        <property name="text">
         <string>Значение</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="lbl_show_size">
        <property name="text">
         <string>Длина</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="lbl_show_p_key">
        <property name="text">
         <string>Ключ Р</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="lbl_p_key_value">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>20</height>
         </size>
        </property>
        <property name="text">
         <string>none</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QLabel" name="lbl_p_key_size">
        <property name="text">
         <string>none</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="lbl_show_s_key">
        <property name="text">
         <string>Ключ S</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLabel" name="lbl_s_key_value">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>20</height>
         </size>
        </property>
        <property name="text">
         <string>none</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QLabel" name="lbl_s_key_size">
        <property name="text">
         <string>none</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="layoutWidget_3">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>304</width>
     <height>58</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="componate_p_key_all">
    <item>
     <layout class="QHBoxLayout" name="componate_p_key">
      <item>
       <layout class="QVBoxLayout" name="componate_p_key_show">
        <item>
         <layout class="QHBoxLayout" name="componate_p_key_show_value">
          <item>
           <widget class="QLabel" name="p_key_show_value">
            <property name="text">
             <string>Ключ Р</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="p_key_show_size">
          <property name="text">
           <string>Длина блока Р</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="componate_p_key_edit_2">
        <item>
         <layout class="QHBoxLayout" name="componate_p_key_edit_value">
          <item>
           <widget class="QLineEdit" name="p_key_edit"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="componate_p_size_edit">
          <item>
           <widget class="QSlider" name="p_key_size_edit_slider">
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="p_key_size_edit_spbox">
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="singleStep">
             <number>1</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QPushButton" name="button_p_key_generate">
      <property name="minimumSize">
       <size>
        <width>50</width>
        <height>50</height>
       </size>
      </property>
      <property name="text">
       <string>Задать</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="layoutWidget_5">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>530</y>
     <width>506</width>
     <height>28</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="componate_algoritm">
    <item>
     <widget class="QLabel" name="lbl_algoritm_show">
      <property name="text">
       <string>Созданный алгоритм шифрования:</string>
      </property>
      <property name="textFormat">
       <enum>Qt::AutoText</enum>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="lbl_algoritm_value">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item>
     <spacer name="horizontalSpacer_4">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>218</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="layoutWidget_6">
   <property name="geometry">
    <rect>
     <x>330</x>
     <y>0</y>
     <width>304</width>
     <height>58</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="componate_s_key_all">
    <item>
     <layout class="QHBoxLayout" name="componate_s_key">
      <item>
       <layout class="QVBoxLayout" name="componate_s_key_show">
        <item>
         <layout class="QHBoxLayout" name="componate_s_key_show_value">
          <item>
           <widget class="QLabel" name="s_key_show_value">
            <property name="text">
             <string>Ключ S</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="s_key_show_size">
          <property name="text">
           <string>Длина блока S</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="componate_s_key_edit">
        <item>
         <layout class="QHBoxLayout" name="componate_s_key_edit_value">
          <item>
           <widget class="QLineEdit" name="s_key_edit"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="componate_s_size_edit">
          <item>
           <widget class="QSlider" name="s_key_size_edit_slider">
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="s_key_size_edit_spbox">
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="singleStep">
             <number>1</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QPushButton" name="button_s_key_generate">
      <property name="minimumSize">
       <size>
        <width>50</width>
        <height>50</height>
       </size>
      </property>
      <property name="text">
       <string>Задать</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>570</y>
     <width>109</width>
     <height>71</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="componate_coordinate">
    <item>
     <widget class="QLabel" name="label_4">
      <property name="text">
       <string>Картинка из позиции</string>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="componate_coordinat_i">
      <item>
       <widget class="QLabel" name="kbk_show_coordinate_i">
        <property name="text">
         <string>i = </string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="edit_coordinate_i">
        <property name="minimumSize">
         <size>
          <width>30</width>
          <height>20</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>30</width>
          <height>20</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_6">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="componate_coordinate_j">
      <item>
       <widget class="QLabel" name="lbl_show_coordinate_j">
        <property name="text">
         <string>j = </string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="edit_coordinate_j">
        <property name="minimumSize">
         <size>
          <width>30</width>
          <height>20</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>30</width>
          <height>20</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_7">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QPushButton" name="button_back">
   <property name="geometry">
    <rect>
     <x>880</x>
     <y>590</y>
     <width>131</width>
     <height>61</height>
    </rect>
   </property>
   <property name="text">
    <string>назад</string>
   </property>
  </widget>
  <widget class="QWidget" name="layoutWidget7">
   <property name="geometry">
    <rect>
     <x>270</x>
     <y>610</y>
     <width>451</width>
     <height>23</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="componate_progress">
    <item>
     <widget class="QLabel" name="lbl_progress">
      <property name="text">
       <string>Обработка</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QProgressBar" name="progress_send">
      <property name="value">
       <number>0</number>
      </property>
      <property name="textVisible">
       <bool>false</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QCheckBox" name="check_box_crypt_keys">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>80</y>
     <width>201</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>Шифровать ключи при передче</string>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>p_key_size_edit_slider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>p_key_size_edit_spbox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>168</x>
     <y>66</y>
    </hint>
    <hint type="destinationlabel">
     <x>208</x>
     <y>63</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>p_key_size_edit_spbox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>p_key_size_edit_slider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>216</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel">
     <x>170</x>
     <y>69</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>s_key_size_edit_spbox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>s_key_size_edit_slider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>536</x>
     <y>57</y>
    </hint>
    <hint type="destinationlabel">
     <x>494</x>
     <y>64</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>s_key_size_edit_slider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>s_key_size_edit_spbox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>505</x>
     <y>56</y>
    </hint>
    <hint type="destinationlabel">
     <x>544</x>
     <y>58</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
